### ISSUE TITLE
Breaking change in persistent-sqlite 2.6.3.2

### ISSUE THREAD

**Issue by bergmark** [yesodweb/persistent#749]
Got this while building the git-annex test suite, was it intentional?

```
Database/Handle.hs:197:44: error:
    • Couldn't match expected type ‘Sqlite.Connection’
                  with actual type ‘Sqlite.Statement’
    • In the first argument of ‘Sqlite.step’, namely ‘stmt’
      In the second argument of ‘($)’, namely ‘Sqlite.step stmt’
      In a stmt of a 'do' block: void $ Sqlite.step stmt
    |
197 |                         void $ Sqlite.step stmt
    |
```

**Comment by paul-rouse** [comment]
Deprecated on hackage:unamused:

@MaxGabriel I think we need to implement #743 with a separate version of `step` for internal use, because `step` itself is visible externally, although undocumented!  I hadn't seen that either when I commented on #730

**Comment by MaxGabriel** [comment]
@bergmark Thanks for finding this. I released persistent-sqlite 2.6.4 which restores the old function signature.

**Comment by bergmark** [comment]
Thanks!
